aiida_mlip/calculations/singlepoint.py

```diff
@@ -176,70 +111,17 @@ def prepare_for_submission(
         aiida.common.datastructures.CalcInfo
             An instance of `aiida.common.datastructures.CalcInfo`.
         """
-        # Create needed inputs
-        # Define architecture from model if model is given,
-        # otherwise get architecture from inputs and download default model
-        architecture = (
-            str((self.inputs.model).architecture)
-            if self.inputs.model
-            else str(self.inputs.architecture.value)
-        )
-        if self.inputs.model:
-            model_path = self.inputs.model.filepath
-        else:
-            model_path = ModelData.download(
-                "https://github.com/stfc/janus-core/raw/main/tests/models/mace_mp_small.model",  # pylint:disable=line-too-long
-                architecture,
-            ).filepath
-
         # The inputs are saved in the node, but we want their value as a string
-        precision = (self.inputs.precision).value
-        device = (self.inputs.device).value
         xyz_filename = (self.inputs.xyz_output_name).value
-        input_filename = self.inputs.metadata.options.input_filename
-        log_filename = (self.inputs.log_filename).value
-
-        # Transform the structure data in xyz file called input_filename
-        structure = self.inputs.structure
 
-        atoms = structure.get_ase()
-        with folder.open(input_filename, "w", encoding="utf-8") as inputfile:
-            write(inputfile, images=atoms)
-
-        cmd_line = {
-            "arch": architecture,
-            "struct": input_filename,
-            "device": device,
-            "log": log_filename,
-            "out": xyz_filename,
-            "calc-kwargs": {"model": model_path, "default_dtype": precision},
-        }
-
-        codeinfo = datastructures.CodeInfo()
-
-        # Initialize cmdline_params as an empty list
-        codeinfo.cmdline_params = []
+        # Call the parent class method to prepare common inputs
+        calcinfo = super().prepare_for_submission(folder)
+        codeinfo = calcinfo.codes_info[0]
 
         # Adding command line params for when we run janus
-        codeinfo.cmdline_params.append("singlepoint")
-        for flag, value in cmd_line.items():
-            codeinfo.cmdline_params += [f"--{flag}", str(value)]
-
-        # Node where the code is saved
-        codeinfo.code_uuid = self.inputs.code.uuid
-        # Save name of output as you need it for running the code
-        codeinfo.stdout_name = self.metadata.options.output_filename
-
-        calcinfo = datastructures.CalcInfo()
-        calcinfo.codes_info = [codeinfo]
-        # Save the info about the node where the calc is stored
-        calcinfo.uuid = str(self.uuid)
-        # Retrieve output files
-        calcinfo.retrieve_list = [
-            self.metadata.options.output_filename,
-            xyz_filename,
-            self.uuid,
-            log_filename,
-        ]
+        codeinfo.cmdline_params[0] = "singlepoint"
```

Comment:
Is this just overwriting the placeholder(?) "calculation"?

Might be worth adding a comment (here and elsewhere), as otherwise it's not immediately obvious.